File: 2020/01/rust/src/main.rs
use itertools::Itertools;
use std::io;

fn main() {
    let mut data = Vec::new();
    for line in io::stdin().lines() {
        data.push(line.unwrap().parse::<i32>().unwrap());
    }

    for numbers in data.iter().combinations(3) {
        if numbers.iter().copied().sum::<i32>() == 2020 {
            println!("{}", numbers.iter().copied().product::<i32>());
        }
    }
}

File: 2020/01/python/main.py
import sys
import itertools
import operator
from functools import reduce


def find_sum(data, N):
    for numbers in itertools.combinations_with_replacement(data, N):
        if sum(numbers) == 2020:
            return reduce(operator.mul, numbers, 1)


data = []
for row in sys.stdin:
    data.append(int(row))

print(find_sum(data, 2))
print(find_sum(data, 3))
